lib/streamlit/hashing.py

```diff
@@ -290,8 +294,10 @@ def _to_bytes(self, obj, context):
                     return self.to_bytes("%s.%s" % (obj.__module__, obj.__name__))
 
                 h = hashlib.new(self.name)
-                # TODO: This may be too restrictive for libraries in development.
-                if os.path.abspath(obj.__code__.co_filename).startswith(os.getcwd()):
+                filepath = os.path.abspath(obj.__code__.co_filename)
+                if filepath.startswith(
+                    os.getcwd()
```

Comment:
I have to think more deeply about this, but just something I noticed: this code uses `cwd` whereas `LocalSourcesWatcher` uses the folder where the main script lives.

Does it make sense to unify the two behaviors? If so, which is best: cwd or main folder?

Reply:
I wonder this is just an artifact of the code being written by two different people. I do think there might be issues with both. What if the main script is in a subdirectory? And what if we execute streamlit from a path that does not include CWD? Should be document and make this a user-option. We can make clear that only a subset of files will be watched/checked for mutation.

I like this to be the same for FileWatcher and here.